napalm/base/helpers.py

```diff
@@ -255,3 +256,47 @@ def as_number(as_number_val):
         return (int(big) << 16) + int(little)
     else:
         return int(as_number_str)
+
+
+def canonical_interface(interface, device_os, change, short=False):
+    '''
+    Function to retun interface canonical name
+    This puposely does not use regex, or first X characters, to ensure
+    there is no chance for false positives. As an example, Po = PortChannel, and
+    PO = POS. With either character or regex, that would produce a false positive.
+    '''
+    if not change:
+        return interface
+
+    if device_os == "cisco_nxos_ssh":
+        device_os = "cisco_nxos"
+    supported_os_list = ["cisco_ios", "cisco_nxos", "cisco_xr", "arista_eos"]
+    if device_os not in supported_os_list:
+        return interface
+
+    def split_on_match(split_interface):
+        '''
+        simple fuction to split on first digit, slash, or space match
+        '''
+        head = split_interface.rstrip(r'/\0123456789 ')
+        tail = split_interface[len(head):].lstrip()
+        return head, tail
+
+    interface_type, interface_number = split_on_match(interface)
+
+    int_map = _interface_map()
+    base_interfaces = int_map['base_interfaces']
+    reverse_mapping = int_map['reverse_mapping']
+    if int_map.get(device_os):
```

Comment:
This should come from the driver itself.

```
class IOSDriver(NetworkDriver):
    # probably needs a better name but this is a class attribute
    long_to_short = {"my_unique_iface_name_that_collides_with_something_else": "ohmy"}
...

def canonical_interface(interface, device, change, short=False):
    ...
    base_interfaces = base_interfaces.update(device.long_to_short)
    ...
```

Reply:
Missed this before.